### PR TITLE
Replace deprecated parse_file

### PR DESCRIPTION
**Pull request by seallard**
## Description
This PR replaces a deprecated function call on a Pydantic model (noticed a warning in my editor).

This syntax was deprecated in Pydantic 2
```python
model.parse_file(json_path)
```

And should be replaced with `model_validate`, see [docs](https://docs.pydantic.dev/2.0/migration/#changes-to-pydanticbasemodel). It was only used in one place in the code base.

### Fixed
- Replace deprecated Pydantic json parsing

### This [version](https://semver.org/) is a
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Review by henrikstranneheim — APPROVED** [review]
👍

File: cg/apps/sequencing_metrics_parser/parsers/bcl2fastq.py [hunk 1]
```diff
@@ -36,7 +37,8 @@ def parse_bcl2fastq_raw_tile_metrics(
 
     for stats_json_path in stats_json_paths:
         LOG.debug(f"Parsing stats.json file {stats_json_path}")
-        sequencing_metrics = Bcl2FastqSampleLaneTileMetrics.parse_file(stats_json_path)
+        data = read_json(stats_json_path)
+        sequencing_metrics = Bcl2FastqSampleLaneTileMetrics.model_validate(data)
```
Comment (islean) [inline cg/apps/sequencing_metrics_parser/parsers/bcl2fastq.py:41]: I think you don't really need to open and read the json, given that there is a model_validate_json function in pydantic v2

```suggestion
        sequencing_metrics = Bcl2FastqSampleLaneTileMetrics.model_validate_json(stats_json_path.open())
```

**Review by islean — APPROVED** [review]
Looks nice!

File: cg/apps/sequencing_metrics_parser/parsers/bcl2fastq.py (hunk 1 above)
Comment (seallard) [reply cg/apps/sequencing_metrics_parser/parsers/bcl2fastq.py:41]: Nice 👍

Comment (seallard) [reply cg/apps/sequencing_metrics_parser/parsers/bcl2fastq.py:41]: Hm, it broke some tests doing it like that. I'll keep it as is since it works

**Pull request merged**